File: core/src/com/lyeeedar/MainGame.java
package com.lyeeedar;

import com.badlogic.gdx.ApplicationAdapter;
import com.badlogic.gdx.Game;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.files.FileHandle;
import com.badlogic.gdx.graphics.GL20;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.lyeeedar.Screens.AbstractScreen;
import com.lyeeedar.Screens.GameScreen;

import javax.swing.*;
import java.io.PrintWriter;
import java.io.StringWriter;
import java.util.HashMap;

public class MainGame extends Game
{
	public static MainGame Instance;

	public MainGame()
	{
		Instance = this;
	}

	public enum ScreenEnum
	{
		GAME
	}

	public final HashMap<ScreenEnum, AbstractScreen> screens = new HashMap<ScreenEnum, AbstractScreen>();

	@Override
	public void create()
	{
		GlobalData.Global.applicationChanger.processResources();
		GlobalData.Global.setup();

		if (!GlobalData.Global.android)
		{
			final StringWriter sw = new StringWriter();
			final PrintWriter pw = new PrintWriter( sw );

			final Thread.UncaughtExceptionHandler handler = new Thread.UncaughtExceptionHandler()
			{
				public void uncaughtException(Thread myThread, Throwable e)
				{
					e.printStackTrace( pw );
					String exceptionAsString = sw.toString();

					FileHandle file = Gdx.files.local( "error.log" );
					file.writeString( exceptionAsString, false );

					JOptionPane.showMessageDialog( null, "An fatal error occured. Please send error.log to me so that I can fix it.", "An error occured", JOptionPane.ERROR_MESSAGE );

					e.printStackTrace();
				}
			};

			Thread.currentThread().setDefaultUncaughtExceptionHandler(handler);
		}

		screens.put( ScreenEnum.GAME, new GameScreen() );

		switchScreen( ScreenEnum.GAME );
	}

	public void switchScreen( ScreenEnum screen )
	{
		this.setScreen( screens.get( screen ) );
	}
}

File: core/src/com/lyeeedar/MainGame.kt
package com.lyeeedar

import com.badlogic.gdx.Game
import com.badlogic.gdx.Gdx
import com.lyeeedar.Screens.AbstractScreen
import com.lyeeedar.Screens.GameScreen
import com.lyeeedar.Screens.ParticleEditorScreen
import java.io.PrintWriter
import java.io.StringWriter
import java.util.*
import javax.swing.JOptionPane

class MainGame : Game()
{
	enum class ScreenEnum
	{
		GAME
	}

	private val screens = HashMap<ScreenEnum, AbstractScreen>()

	override fun create()
	{
		Global.applicationChanger.processResources()
		Global.setup()

		if (Global.android)
		{

		}
		else
		{
			val sw = StringWriter()
			val pw = PrintWriter(sw)

			val handler = Thread.UncaughtExceptionHandler { myThread, e ->
				e.printStackTrace(pw)
				val exceptionAsString = sw.toString()

				val file = Gdx.files.local("error.log")
				file.writeString(exceptionAsString, false)

				JOptionPane.showMessageDialog(null, "A fatal error occurred. Please send the error.log to me so that I can fix it.", "An error occurred", JOptionPane.ERROR_MESSAGE)

				e.printStackTrace()
			}

			Thread.currentThread().uncaughtExceptionHandler = handler
		}

		screens.put(ScreenEnum.GAME, GameScreen())

		if (Global.PARTICLE_EDITOR)
		{
			setScreen(ParticleEditorScreen())
		}
		else
		{
			switchScreen(ScreenEnum.GAME)
		}
	}

	fun switchScreen(screen: AbstractScreen)
	{
		this.setScreen(screen)
	}

	fun switchScreen(screen: ScreenEnum)
	{
		this.setScreen(screens[screen])
	}
}
